networkx/utils/backends.py

```diff
@@ -0,0 +1,628 @@
+                raise TypeError(
+                    f"{name}() graphs must all be from the same plugin, found {plugin_names}"
+                ) from None
+            [plugin_name] = plugin_names
+            if plugin_name not in plugins:
+                raise ImportError(f"Unable to load plugin: {plugin_name}") from None
+            backend = plugins[plugin_name].load()
+            if hasattr(backend, name):
+                return getattr(backend, name).__call__(*args, **kwds)
+            else:
+                raise NetworkXNotImplemented(
+                    f"'{name}' not implemented by {plugin_name}"
+                )
+        return func(*args, **kwds)
+
+    _register_algo(name, wrapper)
+    return wrapper
+
+
+def test_override_dispatch(
+    func=None,
+    *,
+    name=None,
+    graphs="G",
+    edge_attrs=None,
+    node_attrs=None,
+    preserve_edge_attrs=False,
+    preserve_node_attrs=False,
+    preserve_graph_attrs=False,
+):
+    """Auto-converts graph arguments into the backend equivalent,
+    causing the dispatching mechanism to trigger for every
+    decorated algorithm."""
+    if func is None:
+        return functools.partial(
+            _dispatch,
+            name=name,
+            graphs=graphs,
+            edge_attrs=edge_attrs,
+            node_attrs=node_attrs,
+            preserve_edge_attrs=preserve_edge_attrs,
+            preserve_node_attrs=preserve_node_attrs,
+            preserve_graph_attrs=preserve_graph_attrs,
+        )
+    if isinstance(func, str):
+        raise TypeError("'name' and 'graphs' must be passed by keyword") from None
+    # If name not provided, use the name of the function
+    if name is None:
+        name = func.__name__
+
+    if isinstance(graphs, str):
+        graphs = {graphs: 0}
+    elif len(graphs) == 0:
+        raise KeyError("'graphs' must contain at least one variable name") from None
+
+    # This dict comprehension is complicated for better performance; equivalent shown below.
+    optional_graphs = set()
+    list_graphs = set()
+    graphs = {
+        optional_graphs.add(val := k[:-1]) or val
+        if (last := k[-1]) == "?"
+        else list_graphs.add(val := k[1:-1]) or val
+        if last == "]"
+        else k: v
+        for k, v in graphs.items()
+    }
+    # The above is equivalent to:
+    # optional_graphs = {k[:-1] for k in graphs if k[-1] == "?"}
+    # list_graphs = {k[1:-1] for k in graphs if k[-1] == "]"}
+    # graphs = {k[:-1] if k[-1] == "?" else k: v for k, v in graphs.items()}
+
+    sig = inspect.signature(func)
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwds):
+        backend = plugins[plugin_name].load()
+        if not hasattr(backend, name):
+            if plugin_name == "nx-loopback":
+                raise NetworkXNotImplemented(
+                    f"'{name}' not found in {backend.__class__.__name__}"
+                )
+            pytest.xfail(f"'{name}' not implemented by {plugin_name}")
```

Comment:
Would `pytest.skip` help with the coverage? I don't like the idea of calling the NetworkX version. It could easily mask an error or misconfiguration in `graphblas-algorithms` if algorithms aren't being registered correctly with the interface. `xfail` or `skip` feel much safer.

Reply:
I think missing coverage in `graphblas-algorithms` would catch such a mistake.

Reply:
Done. Added `NETWORKX_BACKEND_TEST_EXHAUSTIVE`.